modules/st2-time/time.component.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import Time from 'react-time';
+
+const timeComponent = (props) => {
+  let {format, timestamp, utcdisplay, togglecallback, datatest} = props;
+  let dateFormat = format ? format : 'ddd, DD MMM YYYY HH:mm:ss';
+  let timeStamp = new Date(timestamp);
+  if (utcdisplay){
+    dateFormat += ' UTC';
+  }
+
+  return (
+    <Time
+      data-test={datatest ? datatest : undefined}
```

Comment:
I'd rather you would use {...restProps}. `Data-` props are falling through as is in react.